arangod/Aql/Functions.cpp

```diff
@@ -112,359 +114,399 @@ static_assert(DateSelectionModifier::WEEK < DateSelectionModifier::MONTH,
 static_assert(DateSelectionModifier::MONTH < DateSelectionModifier::YEAR,
               "incorrect date selection order");
 
+typedef void(*format_func_t)(std::string& wrk, tp_sys_clock_ms const&);
+typedef std::unordered_map<std::string,format_func_t> dateMapType;
+std::unordered_map<std::string,format_func_t> dateMap;
+auto unix_epoch = date::sys_seconds{seconds{0}};
 
-/*
-- always specify your user facing function name MYFUNC in error generators
-- errors are broadcasted like this:
-    - Wrong parameter types: RegisterInvalidArgumentWarning(query, "MYFUNC")
-    - Generic errors: RegisterWarning(query, "MYFUNC", TRI_ERROR_QUERY_INVALID_REGEX);
-    - ICU related errors: if (U_FAILURE(status)) { RegisterICUWarning(query, "MYFUNC", status); }
-    - close with: return AqlValue(AqlValueHintNull());
-- specify the number of parameters you expect at least and at max using:
-  ValidateParameters(parameters, "MYFUNC", 1, 3); (min: 1, max: 3); Max is optional.
-- if you support optional parameters, first check whether the count is sufficient
-  using parameters.size()
-- fetch the values using:
-  AqlValue value
-  - Anonymous  = ExtractFunctionParameterValue(parameters, 0);
-  - GetBooleanParameter() if you expect a bool
-  - Stringify() if you need a string.
-  - ExtractKeys() if its an object and you need the keys
-  - ExtractCollectionName() if you expect a collection
-  - ListContainsElement() search for a member
-  - ParameterToTimePoint / DateFromParameters get a time string as date.
-- check the values whether they match your expectations i.e. using:
-  - param.isNumber() then extract it using: param.toInt64(trx)
-- Available helper functions for working with pamaterers:
-  - Variance()
-  - SortNumberList()
-  - UnsetOrKeep ()
-  - GetDocumentByIdentifier ()
-  - MergeParameters()
-  - FlattenList()
+std::regex theDateFormatRegex;
 
-- now do your work with the parameters
-- build ub a result using a VPackbuilder like you would with regular velocpyack.
-- return it wrapping it into an AqlValue
+std::string regexReplace(std::string const& search,
```

Comment:
If we want to keep the generic name, we should move this out to i.e. basics, and replace the `dateMapType` with some generic name, and make tp a template parameter.
Else drop this generic name for a rather special purpose one.